src/js/themes/media-theme-template.js

```diff
@@ -0,0 +1,117 @@
+import { defineCustomElement } from '../utils/defineCustomElement.js';
+import MediaTheme from './media-theme.js';
+import { InnerTemplatePart, TemplateInstance, AttrPart } from './template-parts.js';
+
+class MediaThemeTemplate extends MediaTheme {
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    this.template = this.getRootNode().querySelector(`#${this.getAttribute('template')}`);
+
+    this.templateInstance = new TemplateInstance(
+      this.template,
+      this.props,
+      mediaThemeProcessor
+    );
+
+    this.shadowRoot.append(this.templateInstance);
+
+    const observer = new MutationObserver(() => this.render());
+    observer.observe(this, { attributes: true });
+  }
+
+  get props() {
+    const props = {};
+    for (let attr of this.attributes) {
+      if (attr.value != null) {
+        props[camelCase(attr.name)] = attr.value === '' ? true : attr.value;
+      } else {
+        props[camelCase(attr.name)] = false;
+      }
+    }
+    return props;
+  }
+
+  render() {
+    this.templateInstance.update(this.props);
+  }
+}
+
+const camelCase = (name) => {
+  return name.replace(/[-_]([a-z])/g, ($0, $1) => $1.toUpperCase());
+};
+
+const mediaThemeProcessor = {
+  processCallback(instance, parts, state) {
+    if (!state) return;
+    for (const [expression, part] of parts) {
+      if (part instanceof InnerTemplatePart) {
+        console.log(part);
+        switch (part.directive) {
+          case 'set': {
+            // cache these so they return the same element across layouts
```

Comment:
That means the element can only be used once per layout, which is different from a _partial_ (or just any _template_) which could be used multiple times in the same layout. I can see value in sharing the element in some cases, but i'm not sure that should be the default behavior of a sub-template/partial. I think that's probably what you're getting at by using the term `set`, so I think I get it.

Reply:
yes maybe it's not the best idea, users will have to know the same element can't be in 2 places in the dom.
we can leave caching but it will clone the element each time, maybe it's fine

Reply:
If there's value in creating a reference to the same element I think we can do both, we just need to be real clear with the naming of the directives between what's meant to be used as a _template partial_ and what's meant to be a _reference_.